integrations/ultralytics/server/server.py

```diff
@@ -0,0 +1,267 @@
+
+positional arguments:
+  onnx_filepath         The full filepath of the ONNX model file or SparseZoo stub
+                        to the model
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -s BATCH_SIZE, --batch_size BATCH_SIZE
+                        The batch size to run the analysis for
+  -j NUM_CORES, --num_cores NUM_CORES
+                        The number of physical cores to run the analysis on,
+                        defaults to all physical cores available on the system
+  -a ADDRESS, --address ADDRESS
+                        The IP address of the hosted model
+  -p PORT, --port PORT  The port that the model is hosted on
+
+##########
+Example command for running:
+python server.py \
+    ~/models/yolo-v3-pruned_quant.onnx
+"""
+
+import argparse
+import time
+from typing import List, Tuple
+
+import numpy
+import torch
+
+import cv2
+import flask
+from deepsparse import compile_model
+from flask_cors import CORS
+
+# ultralytics/yolov5 imports
+from utils.general import non_max_suppression
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Host a Yolo ONNX model as a server, using the DeepSparse Engine and Flask"
+        )
+    )
+
+    parser.add_argument(
+        "onnx_filepath",
+        type=str,
+        help="The full filepath of the ONNX model file or SparseZoo stub to the model",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--batch_size",
+        type=int,
+        default=1,
+        help="The batch size to run the analysis for",
+    )
+    parser.add_argument(
+        "-j",
+        "--num_cores",
+        type=int,
+        default=0,
+        help=(
+            "The number of physical cores to run the analysis on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-a",
+        "--address",
+        type=str,
+        default="0.0.0.0",
+        help="The IP address of the hosted model",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        type=str,
+        default="5543",
+        help="The port that the model is hosted on",
+    )
+
+    return parser.parse_args()
+
+
+def _get_grid(size: int) -> torch.Tensor:
+    # adapted from yolov5.yolo.Detect._make_grid
+    coords_y, coords_x = torch.meshgrid([torch.arange(size), torch.arange(size)])
+    grid = torch.stack((coords_x, coords_y), 2)
+    return grid.view(1, 1, size, size, 2).float()
+
+
+# Yolo V3 specific variables
+_YOLO_V3_ANCHORS = [
+    torch.Tensor([[10, 13], [16, 30], [33, 23]]),
+    torch.Tensor([[30, 61], [62, 45], [59, 119]]),
+    torch.Tensor([[116, 90], [156, 198], [373, 326]]),
+]
+_YOLO_V3_ANCHOR_GRIDS = [t.clone().view(1, -1, 1, 1, 2) for t in _YOLO_V3_ANCHORS]
+_YOLO_V3_OUTPUT_SHAPES = [80, 40, 20]
+_YOLO_V3_GRIDS = [_get_grid(grid_size) for grid_size in _YOLO_V3_OUTPUT_SHAPES]
+
+
+def arrays_to_bytes(arrays: List[numpy.array]) -> str:
```

Comment:
is this something more general we should add as a utility to sparseml?

Reply:
added here: https://github.com/neuralmagic/deepsparse/pull/62